_data/upcoming_events.yml

```diff
@@ -0,0 +1,15 @@
+# Contribute
```

Comment:
It might be good to somehow integrate this with the OMP Calendar, just to make maintenance easier...

https://www.openmainframeproject.org/events/category/zowe

Reply:
Would like to learn how the events are added in [OMP website](https://www.openmainframeproject.org/events/category/zowe) and then can try to sync it with zowe.org. Is there any open-souced repository for [OMP website](https://www.openmainframeproject.org/events/category/zowe)?

Reply:
We add the events as folks let us know, and the tag them appropriately. You can pull an ICS feed ( https://www.openmainframeproject.org/events/category/zowe?ical=1&tribe_display=photo )and parse into this YAML format - probably a simple script to throw together ( which I could help out with if you'd like )

Reply:
I think we will need to convert the ICS (iCalendar) to JSON first and then we can parse it on the webage?

Reply:
YAML - but seeing what you are needing should be fairly trivial to create a script and run as a GH action.

Reply:
I tried fetching the response and it returns [vObject/vFormat](https://tools.ietf.org/id/draft-calconnect-vobject-vformat-00.html) data. I'm not really sure how to parse it to YAML. It would be really helpful if you can help me with the script.

Reply:
A script like this should work....

```python
#!/usr/bin/env python3
#
# Copyright this project and it's contributors
# SPDX-License-Identifier: Apache-2.0
#
# encoding=utf8

from ics import Calendar
import requests
import yaml

url = 'https://www.openmainframeproject.org/events/category/zowe?ical=1&tribe_display=photo'
c = Calendar(requests.get(url).text)

eventlist = []
for event in list(c.events):
    eventinstance = {}
    eventinstance['event'] = event.name
    eventinstance['start_date'] = event.begin.format()
    eventinstance['end_date'] = event.end.format()
    eventinstance['description'] = event.description
    eventinstance['url'] = event.url
    eventlist.append(eventinstance)

with open(r'../_data/upcoming_events.yml', 'w') as yamlfile:
    yaml.dump(eventlist,yamlfile, sort_keys=False)
```

Reply:
Thank you, it worked. Also when shall the script run, should it be _within_ the Jekyll build or before like `python assets/script/ics-to-yml.py && bundle exec jekyll build`

Reply:
There's probably a lot of ways to approach this, but one that might work is a GH actions job that runs as a cron job. Here's an example of how that could work in what I do for the landscape...

https://github.com/openmainframeproject/omp-landscape/blob/master/.github/workflows/update_members.yml

Also, this script should be something that doesn't get built by jekyll, so you might want to ensure wherever it is that it doesn't get added.

Reply:
Hey, can you please review it. I added a [GH action](https://github.com/covalentbond/zowe.github.io/blob/6c3dc4a7bf9461a433c02078582a0567916abd78/.github/workflows/update_upcoming_events.yml) to trigger the script once every day and will send PR only when the contents of [upcoming_events.yml](https://github.com/covalentbond/zowe.github.io/blob/6c3dc4a7bf9461a433c02078582a0567916abd78/assets/script/ics-to-yml.py) will change.

You might also need to add `PAT` [Personal access token](https://docs.github.com/en/github/authenticating-to-github/creating-a-personal-access-token) for the Pull request feature

Reply:
Resolving this for now, will require to test the above GitHub Action once it's merged and check if it's causing any error.